src/gui_common/CustomPopupMenu.cs

```diff
@@ -44,33 +42,26 @@ public void RemapDynamicChildren()
         }
     }
 
-    public override void Open()
+    protected virtual void ResolveNodeReferences()
     {
-        base.Open();
+    }
 
+    protected override void OnOpen()
+    {
         CreateTween().TweenProperty(this, "rect_size", cachedMinSize, 0.2f)
             .From(Vector2.Zero)
             .SetTrans(Tween.TransitionType.Circ)
             .SetEase(Tween.EaseType.Out);
     }
 
-    public override void Close()
+    protected override void OnClose()
     {
-        if (closing || !Visible)
-            return;
-
-        closing = true;
-
         var tween = CreateTween();
         tween.TweenProperty(this, "rect_size", Vector2.Zero, 0.15f)
             .From(cachedMinSize)
             .SetTrans(Tween.TransitionType.Circ)
             .SetEase(Tween.EaseType.Out);
-        tween.TweenCallback(this, nameof(OnFolded));
-    }
-
-    protected virtual void ResolveNodeReferences()
-    {
+        tween.TweenCallback(this, "hide");
```

Comment:
Is this intentional that this now calls "hide" instead of `nameof(Hide)`?

Reply:
Yes. I can't find the link to the Godot docs reference right now but I think it stated that you must use the native name when calling engine functions like this.

Reply:
If that's the intention, that is correct, but reading the existing code, this used to send the Close method call to our code. Why the change from that to the godot inbuilt method?

Reply:
Because we no longer have to set `closing` to false, `OnFolded` isn't needed, one statement only.

Reply:
> Or is this an animation specific use case where the hide needs to be called?

Kind of, or any other thing you'd want the window to do when you call `Close` to it really.

https://github.com/Revolutionary-Games/Thrive/blob/4cb9670399c902dd7ebea2feb04bbbe543ef94a0/src/gui_common/CustomWindow.cs#L225-L238

> "OnHideAnimationComplete"

Already covered functionally by `OnHidden`.

Reply:
Then why does this not call that `OnHidden`?

My point here is that a custom functionality we are adding to the windows, should use our own methods to control the program flow, not rely on calling a Godot inbuilt method which then through more complicated callbacks eventually does the right thing. That's my point that I'd much prefer to still call one of our own methods here rather than "hide".

Reply:
Okay.